model/vm.rb

```diff
@@ -26,6 +26,19 @@ def hyper_tag_name(project)
 
   include Authorization::TaggableMethods
 
+  DISPLAY_STATES = {
+    start: "creating",
+    create_unix_user: "creating",
+    prep: "creating",
+    trigger_refresh_mesh: "creating",
+    run: "creating",
+    wait_sshable: "creating",
+    wait: "running",
+    refresh_mesh: "running",
+    destroy: "deleting",
+    start_after_host_reboot: "starting"
+  }
```

Comment:
How do you ensure this is exhaustive? Is it better to update a display_state field in those relevant labels? Then you don't have to say one state after another is "creating"

Reply:
Engineers who work generally on backend side forget to update display state for their newly added labels. With this list, we enforce to add display state for new labels.

But yeah I'm not big fan to lots of duplicate display states too.

Maybe keeping display state column on DB than update when it's changed
(I used UBID.decode to get resource of nexus)
```
# Keep only changed states on this map
 DISPLAY_STATES = {
    start: "creating",
    wait: "running",
    destroy: "deleting",
    start_after_host_reboot: "starting"
  }

# Update display_state if it has value while hopping
def dynamic_hop(label)
    fail "BUG: #hop only accepts a symbol" unless label.is_a? Symbol
    fail "BUG: not valid hop target" unless self.class.labels.include? label
    if (new_state = self.class::DISPLAY_STATES[label])
      UBID.decode(id).update(display: new_state)
    end
    label = label.to_s
    fail Hop.new(@strand.prog, @strand.label, {label: label, retval: nil})
end

# Or maybe move to where we update strand too
def unsynchronized_run
.
.
.
    rescue Prog::Base::Hop => hp
      update(**hp.strand_update_args)
      if (new_state = self.class::DISPLAY_STATES[hp.new_label])
        UBID.decode(id).update(display: new_state)
      end
      hp
    rescue Prog::Base::Exit ⇨ ext
.
.
.
end
```